python/ray/data/_internal/logging.py

```diff
@@ -0,0 +1,64 @@
+import logging
+import os
+from typing import Optional
+
+import ray
+from ray._private.ray_constants import LOGGER_FORMAT
+
+_default_log_handler: logging.Handler = None
+
+DEFAULT_DATASET_LOG_FILENAME = "ray-data.log"
+
+
+class LogHandler(logging.Handler):
+    def __init__(self, filename: str):
+        super().__init__()
+        self._filename = filename
+        self._initialized = False
+        self._handler = None
+        self._formatter = None
+        self._path = None
+
+    def emit(self, record):
+        if not self._initialized:
+            self._initialize_handler()
+        if self._handler is not None:
+            self._handler.emit(record)
+
+    def setFormatter(self, fmt: logging.Formatter) -> None:
+        self._formatter = fmt
+
+    def _initialize_handler(self):
+        global_node = ray._private.worker._global_node
+        if global_node is None:
+            self._handler = None
+        else:
+            session_dir = global_node.get_session_dir_path()
+            self._path = os.path.join(session_dir, "logs", self._filename)
+            self._handler = logging.FileHandler(self._path)
+            if self._formatter is not None:
+                self._handler.setFormatter(self._formatter)
+        self._initialized = True
+
+    @property
+    def path(self) -> Optional[str]:
+        return self._path
+
+
+def configure_logging():
```

Comment:
I know this is part of the private API but it would be useful to have some docs here. Even something simple like "Configure the Ray Data logging system. Should be called only once in `<whatever module>`; will fail if called a second time." I know my LSP at least will show this kind of information when I hover over a symbol, and it could give useful context.